index.js

```diff
@@ -136,61 +136,104 @@ function RpcAgent(opts = {}) {
   this.user = opts.user || 'user';
   this.pass = opts.pass || 'pass';
   this.prot = opts.ssl ? https : http;
+  this.calb = false; // Defaults to promises
 }
 
 function rpc(request, callback) {
-  const requestSerialized = JSON.stringify(request);
-  const auth = new Buffer(`${this.user}:${this.pass}`).toString('base64');
-  const options = {
-    host: this.host,
-    port: this.port,
-    path: '/',
-    method: 'POST',
-  };
-  let err = null;
-  const req = this.prot.request(options, (res) => {
-    let buf = '';
-    res.on('data', (data) => {
-      buf += data;
+  return new Promise((resolve, reject) => {
+    const requestSerialized = JSON.stringify(request);
+    const auth = new Buffer(`${this.user}:${this.pass}`).toString('base64');
+    const options = {
+      host: this.host,
+      port: this.port,
+      path: '/',
+      method: 'POST',
+    };
+    let err = null;
+    const req = this.prot.request(options, (res) => {
+      let buf = '';
+      res.on('data', (data) => {
+        buf += data;
+      });
+      res.on('end', () => {
+        if (res.statusCode === 401) {
+          if(this.calb) {
+            return callback(new Error('bitcoin JSON-RPC connection rejected: 401 unauthorized'));
+          }
+          reject(new Error('bitcoin JSON-RPC connection rejected: 401 unauthorized'));
+          return
```

Comment:
It's perfectly fine to return the reject call here and in the other places, I think, to save the extra return lines.